tests/sources/subs/HTML2Md.class.Basic.php

```diff
@@ -30,7 +30,7 @@ public function setUp()
 			array(
 				'Named links',
 				'<a href="http://www.elkarte.net/" class="bbc_link" target="_blank">ElkArte</a>',
-				'[ElkArte](http://www.elkarte.net/)',
+				'[ElkArte](http://www.elkarte.net/ 🔗)',
```

Comment:
:raised_eyebrow:

Reply:
This is a trick ... and maybe not the best, but here is the long story.

When we convert links to markdown for email output the final ')' of the ```[]()``` syntax can cause problems with email clients .. so if your link ended in a ) ,which is a valid link character, then when converted to MD the email client will see it as a )) and  boom bad link.  That is just one example of rouge links.

Adding in that utf8 character forces our outbound email to also output a base64 version (since there is a non ASCII character) in addition to the plain text version.  The base64 version is much less susceptible to getting munged up in an email client (specifically wrong text wrapping) and email clients will prefer to show that over the plain text.  If you look at the plain text you would just see a &#1234 thingy at the end but the link would still work.  It only does this trick when needed.  We could just add a space and the link would then work, as long as it did not get wrapped by the client, but we end up loosing the base64 version in the email

I also fixed the wrong calculation of line wrapping so at least in the plain text version we don't break the link, however some email clients do this on their own.

Reply:
Oooohhh... cool!
Thanks for the explanation!